src/wristpy/processing/calibration.py

```diff
@@ -0,0 +1,430 @@
+
+        This method implements an iterative algorithm that finds the optimal scale and
+        offset for calibrating accelerometer data. The ndarray it processes contains the
+        periods of "no motion" identified by the _extract_no_motion method. This data is
+        then fit to an idealized unit sphere which represents the accelerometer data
+        under noiseless conditions. The scale and offset derived from this process are
+        used to calibrate our data along each axis.
+
+        Args:
+            no_motion_data: The acceleration data during periods of no
+                motion, in order to determine scale and offset.
+
+        Returns:
+            A LinearTransformation object with scale and offset attributes to be applied
+            to acceleration data for calibration.
+
+        Raises:
+            SphereCriteriaError: If the sphere is not sufficiently populated, i.e. every
+                axis does not have at least 1 value both above and below  the + and
+                - value of min_acceleraiton.
+            ZeroScaleError: Numerical instability inherent to the algorithm may cause
+                values of the scale vector to tend toward zero under certain conditions.
+                This will cause calibration to fail.
+
+        References:
+            van Hees VT, Fang Z, Langford J, et al. Autocalibration of accelerometer
+            data for free-living physical activity assessment using local gravity
+            and temperature: an evaluation on four continents. J Appl Physiol (1985)
+            2014 Oct 1;117(7):738-44. doi: 10.1152/japplphysiol.00421.2014.
+        """
+        sphere_criteria_check = np.all(
+            (no_motion_data.min(axis=0) < -self.min_acceleration)
+            & (no_motion_data.max(axis=0) > self.min_acceleration)
+        )
+        if not sphere_criteria_check:
+            raise SphereCriteriaError(
+                "Did not meet criteria to sufficnetly populate sphere"
+            )
+
+        weights = np.ones(no_motion_data.shape[0]) * 100
+        previous_residual = np.Inf
+
+        linear_regression_model = linear_model.LinearRegression()
+
+        offset = np.zeros(3)
+        scale = np.ones(3)
+        for _ in range(self.max_iterations):
+            current = (no_motion_data * scale) + offset
+            closest_point = current / np.linalg.norm(current, axis=1, keepdims=True)
+            linear_regression_model.fit(current, closest_point, sample_weight=weights)
+            offset_change = linear_regression_model.intercept_
+            scale_change = np.diag(linear_regression_model.coef_)
+
+            if np.all((scale * scale_change) == 0):
+                raise ZeroScaleError(
+                    """Calibration has failed as a result of zero scale values."""
+                )
+
+            scale *= scale_change
+            offset += offset_change / scale
+
+            residual = 0.03 * metrics.mean_squared_error(
```

Comment:
I was a bit confused here because our `metrics` doesn't have a mean_squared_error function, and then I realized that it came from `sklear.metrics.mean_squared_error`. Not sure if it is something to be careful with.

Reply:
That's a good point, is it more important to stick to our naming convention or make it clear that this is sklearn.metrics and not wristpy.processings.metrics? @ReinderVosDeWael

Reply:
I refer you to the style guide https://google.github.io/styleguide/pyguide.html#224-decision